File: 05-12-2024_Move_Pieces_to_Obtain_a_String.java
class Solution {
    public boolean canChange(String start, String target) {
        int i = 0, j = 0;
        int n = start.length();
        start += 'x'; target += 'x'; //to mark the end of the string
        while (i < n || j < n) {
            while (i < n && start.charAt(i) == '_') i++;
            while (j < n && target.charAt(j) == '_') j++;
            if (start.charAt(i) != target.charAt(j)) return false;
            if (start.charAt(i) == 'L' && j > i) return false;
            if (start.charAt(i) == 'R' && j < i) return false;
            i++; j++;
        }
        return true;
    }
}

File: 05-12-2024_Move_Pieces_to_Obtain_a_String.py
class Solution:
    def canChange(self, start: str, target: str) -> bool:
        i = 0
        j = 0
        n = len(start)
        start += 'x' #to mark the end of the string
        target += 'x' 
        while i < n or j < n:
            while i < n and start[i] == '_':
                i += 1
            while j < n and target[j] == '_':
                j += 1
            if start[i] != target[j]:
                return False
            if start[i] == 'L' and j > i:
                return False
            if start[i] == 'R' and j < i:
                return False
            i += 1
            j += 1
        return True
